src/main/scala/it/pagopa/interop/authorizationmanagement/model/persistence/KeyAdapters.scala

```diff
@@ -1,23 +1,29 @@
 package it.pagopa.interop.authorizationmanagement.model.persistence
 
+import com.nimbusds.jose.jwk._
 import it.pagopa.interop.authorizationmanagement.errors.KeyManagementErrors.ThumbprintCalculationError
 import it.pagopa.interop.authorizationmanagement.model.key.{Enc, PersistentKey, PersistentKeyUse, Sig}
-import it.pagopa.interop.authorizationmanagement.model.{ClientKey, KeyUse}
-import it.pagopa.interop.authorizationmanagement.service.impl.KeyProcessor
+import it.pagopa.interop.authorizationmanagement.model.{ClientKey, KeyUse, Key, OtherPrimeInfo}
+import it.pagopa.interop.authorizationmanagement.processor.key.KeyProcessor
+
+import scala.annotation.nowarn
+import scala.jdk.CollectionConverters.{ListHasAsScala, SetHasAsScala}
 
 object KeyAdapters {
 
   implicit class PersistentKeyWrapper(private val p: PersistentKey) extends AnyVal {
     def toApi: Either[Throwable, ClientKey] =
-      KeyProcessor
-        .fromBase64encodedPEMToAPIKey(p.kid, p.encodedPem, p.use, p.algorithm)
+      fromBase64encodedPEMToAPIKey(p.kid, p.encodedPem, p.use, p.algorithm)
         .map(ClientKey(_, p.relationshipId, p.name, p.createdAt))
   }
 
   implicit class PersistentKeyObjectWrapper(private val p: PersistentKey.type) extends AnyVal {
     def toPersistentKey(validKey: ValidKey): Either[ThumbprintCalculationError, PersistentKey] =
       for {
-        kid <- KeyProcessor.calculateKid(validKey._2)
+        kid <- KeyProcessor
+          .calculateKid(validKey._2)
+          .left
+          .map(ex => ThumbprintCalculationError(ex.getLocalizedMessage()))
```

Comment:
IMO you can move this leftMap inside the `calculateKid`, that will return an `Either[ThumbprintCalculationError, String]`